### PR TITLE
TST: add messages to pytest.raises

### PR DESCRIPTION
**Pull request by proost**
- [x] xref #30999
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/arrays/boolean/test_function.py [hunk 1]
```diff
@@ -54,7 +54,8 @@ def test_ufuncs_binary(ufunc):
     tm.assert_extension_array_equal(result, expected)
 
     # not handled types
-    with pytest.raises(TypeError):
+    msg = r"operand type\(s\) all returned NotImplemented from \_\_array\_ufunc\_\_"
```
Comment (simonjayhawkins) [inline pandas/tests/arrays/boolean/test_function.py:57]: why are the underscores escaped?

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @proost generally lgtm.

File: pandas/tests/arrays/boolean/test_function.py (hunk 1 above)
Comment (proost) [reply pandas/tests/arrays/boolean/test_function.py]: @simonjayhawkins 
Oh, it's my fault. I fix it.

**Review by simonjayhawkins — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @proost

**Pull request merged**